nav-app/src/app/datatable/data-table.component.ts

```diff
@@ -285,13 +285,26 @@ export class DataTableComponent implements AfterViewInit {
     /**
      * Show all sub-techniques in layout view
      */
-    expandSubtechniques(): void {
+    expandSubtechniques(showAnnotatedOnly?: boolean): void {
         if (this.viewModel.layout.layout == "mini") return; //control disabled in mini layout
         for (let technique of this.dataService.getDomain(this.viewModel.domainID).techniques) {
             if (technique.subtechniques.length > 0) {
                 for (let id of technique.get_all_technique_tactic_ids()) {
                     let tvm = this.viewModel.getTechniqueVM_id(id);
-                    tvm.showSubtechniques = true;
+                    if (!showAnnotatedOnly) {
+                        tvm.showSubtechniques = true;
+                    } else {
+                        let foundAnnotated = false;
+                        for (let subtechnique of technique.subtechniques) {
+                            for (let sid of subtechnique.get_all_technique_tactic_ids()) {
+                                let svm = this.viewModel.getTechniqueVM_id(sid);
+                                if (svm.annotated() && !foundAnnotated) {
+                                    foundAnnotated = true; // set boolean after it finds the first annotated subtechnique
+                                }
+                                tvm.showSubtechniques = foundAnnotated;  // otherwise, keep going and collapse if no annotations found
```

Comment:
This actually isn't the expected behavior. My expectation, according to the iconography and behavior of the counterpart "expand all" control, is that the button wouldn't affect techniques that were already expanded. 

The fact that it collapses all unannotated techniques isn't quite what I was expecting here, and limits the user (they may want to keep their current set of expanded techniques and also expand ones with annotations). The functionality of only expanding ones with annotations can be replicated by collapsing all and then clicking the button, so we don't really gain any functionality from this behavior.

The fix would be to make this line into `tvm.showSubtechniques == tvm.showSubtechniques || foundAnnotated` so that it keeps its expanded state it it was already set.